configure

```diff
@@ -15,8 +15,15 @@ function abort { >&2 echo -e "\033[1m\033[31m$1\033[0m"; exit 1; }
 function info { >&2 echo -e "\033[1m\033[33m$1\033[0m"; }
 function warn { >&2 echo -e "\033[1m\033[33m$1\033[0m"; }
 
-# Install mason
-export PATH="`pwd`/.mason:${PATH}" MASON_DIR="`pwd`/.mason"
+if [ -d "`pwd`/.git" ]; then
+    info "This build is within a git repository"
+    export MASON_DIR="`pwd`/.mason"
+    export PATH="${MASON_DIR}:${PATH}"
+else
+    info "This build is NOT within a git repository"
+    which mason || abort "You must install mason to build mapbox-gl-native (https://github.com/mapbox/mason)"
+    export MASON_DIR="$(dirname $(readlink $(which mason)))"
```

Comment:
@mikemorris by "this" I was mainly referring to `$(dirname $(readlink $(which mason)))`